framework/src/org/checkerframework/framework/util/MultiGraphQualifierHierarchy.java

```diff
@@ -236,10 +236,28 @@ protected void finish(QualifierHierarchy qualHierarchy,
     public String toString() {
         // TODO: it would be easier to debug if the graph and map were sorted by the key.
         // Simply creating a TreeMap here doesn't work, because AnnotationMirrors are not comparable.
-        return "Supertypes Graph: " + supertypesGraph.toString() +
-                "\nSupertypes Map: " + String.valueOf(supertypesMap) +
-                "\nTops: " + tops +
+
+        String output = "Supertypes Graph: ";
+        for (AnnotationMirror key : supertypesGraph.keySet()) {
```

Comment:
I would iterate through the entrySet, because you will access key and value in the body.